packages/core/src/render3/instructions/defer.ts

```diff
@@ -472,30 +472,31 @@ function registerDomTrigger(
     callback: VoidFunction) {
   const injector = initialLView[INJECTOR]!;
 
-  // Assumption: the `afterRender` reference should be destroyed
-  // automatically so we don't need to keep track of it.
-  const afterRenderRef = afterRender(() => {
+  function pollDomTrigger() {
+    // If the initial view was destroyed, we don't need to do anything.
+    if (initialLView[FLAGS] & LViewFlags.Destroyed) {
```

Comment:
nit: there is a helper function that we can use here (and there is a similar check below that we can refactor):
```suggestion
    if (isDestroyed(initialLView)) {
```